### PR TITLE
SI-6946, SI-6924 Greatly improves IsTraversableLike docs

### PR DESCRIPTION
**Pull request by heathermiller**
Given:
1. SI-6924 Error in IsTraversableLike scaladoc example (example code did not compile)
2. SI-6946 Documentation for IsTraversableLike is confusing (my dissatisfaction with the docs for IsTraversableLike)

I rewrote the docs completely and provided an alternate example. While the fix provided in #1850 did indeed make the example code compile, the example code was still far too complicated to clearly illustrate an already confusing concept.

Funny observation: the amount of docs now greatly exceeds the amount of code.

Unfortunately, this renders PR #1850 now unnecessary.

Review by @jsuereth and @phaller

### CODE REVIEW

File: src/library/scala/collection/generic/IsTraversableLike.scala [hunk 1]
```diff
@@ -9,26 +9,97 @@
...
+ *  extension methods that work both on collection types and on `String`s (`String`s
+ *  do not extend `GenTraversableLike`, but can be converted to `GenTraversableLike`)
  *
- *  This type enables simple enrichment of `GenTraversable`s with extension
- *  methods which can make full use of the mechanics of the Scala collections
- *  framework in their implementation.
+ * `IsTraversable` provides two members:
+ *
+ *  1. type member `A`, which represents the element type of the target `GenTraversableLike[A, Repr]`
+ *  1. value member `conversion`, which provides a way to convert between the type we wish to add extension methods to, `Repr`, and `GenTraversableLike[A, Repr]`.
```
Comment (jsuereth) [inline src/library/scala/collection/generic/IsTraversableLike.scala:24]: Should probably be 2.

**Comment by jsuereth** [comment]
Wow, this is amazing heather.   Great work!   Did you want to update IsTraversableOnce as well?   Also, you should think about writing books :-) 

Lgtm

File: src/library/scala/collection/generic/IsTraversableLike.scala (hunk 1 above)
Comment (heathermiller) [inline src/library/scala/collection/generic/IsTraversableLike.scala:24]: Looks like it should be, right? ...Actually no-- Scaladoc's parser has some quirks. The only way I can get it to properly output an ordered list is by doing it this way, all numbered elements have to be 1.

Comment (heathermiller) [inline src/library/scala/collection/generic/IsTraversableLike.scala:24]: Scaladoc sanitzes HTML, and only allows through certain tags. ol is not one of them, so I can't even do it manually. Silly, eh?

Comment (retronym) [inline src/library/scala/collection/generic/IsTraversableLike.scala:24]: Can you raise a bug? We should allow manual numberings...

Comment (jsuereth) [inline src/library/scala/collection/generic/IsTraversableLike.scala:24]: Wow!   Yeah.... I stand corrected :-)

Comment (heathermiller) [inline src/library/scala/collection/generic/IsTraversableLike.scala:24]: @retronym Sure-- there might one one already in existence. In any case, I'll take a look.

**Comment by phaller** [comment]
@jsuereth I agree, I think it reads really well. LGTM

**Comment by soc** [comment]
Looks very very good to me. Great work Heather!

**Comment by milessabin** [comment]
What Josh said .... good work :-)

**Pull request merged**